dotrix_core/src/world.rs

```diff
@@ -61,7 +64,7 @@ impl World {
     /// _NOTE: if your entity has only one component, don't forget to put trailing comma after it
     /// as it is shown in the example above. Otherwise it won't be parsed by Rust compiler as
     /// a tuple.
-    pub fn spawn<T, I>(&mut self, iter: I)
+    pub fn spawn<T, I>(&mut self, iter: I) -> &[Entity]
```

Comment:
Why as a slice rather than an owned `Vec`? Entities are trivially copyable, then we won't need the `spawned` buffer in the struct either.

Reply:
I had it as a `Vec<Entitny>` in the beginning. But I can hardly pretend a use case when you will take this returned data and store it somewhere as is. In many cases you will spawn one entity and store only its id somewhere. As you said it is trivially copyable. 

In some cases (as I  had in terrain) I would iterate over them and store only the id in each appropriate structure (tile structure). 

In some cases you even don't care about the id. 

Returning of `Vec` will require at least one memory allocation that may require also a relatively expensive system call.

So I decided to keep a buffer for return values inside of the world and return it as a slice. 

Another option is to return an Iterator. And as more I think of it, more I like it. And if you want to have them in a `Vec`, you can collect them.

Reply:
So now it also possible to get first, last, number of entities and effectively convert the Iterator into a vector:
```
    #[test]
    fn spawn_and_check_entities() {
        let mut world = World::default();
        let mut spawned = world.spawn(Some((Armor(1),)));

        assert_eq!(spawned.next(), Some(Entity::from(0)));
        assert_eq!(spawned.first(), Entity::from(0));
        assert_eq!(spawned.last(), Some(Entity::from(0)));

        let spawned: Vec<Entity> = world
            .spawn([
                (Armor(2), Health(100)),
                (Armor(4), Health(80)),
                (Armor(4), Health(90)),
            ])
            .into();

        assert_eq!(spawned.len(), 3);

        for i in 0..3 {
            assert_eq!(spawned[i], Entity::from(i as u64 + 1));
        }
    }
    ```

Reply:
That is quite cool :). One last thing since we know the exact size of the Iter we could also implement the [`ExactSizeIterator`](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html#)

Reply:
Good point. Done.